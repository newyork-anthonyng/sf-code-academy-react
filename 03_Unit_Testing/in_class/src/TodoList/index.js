import React from "react";
import PropTypes from "prop-types";
import TodoItem from "../TodoItem";

const TodoList = function(props) {
  return (
    <ul>
      {
        props.todoItems.map((todoItem, index) => (
          <TodoItem
            key={index}
            text={todoItem.text}
            index={index}
            onClick={props.onTodoItemClick}
            completed={todoItem.completed}
          />
        ))
      }
    </ul>
  );
};

TodoList.defaultProps = {
  todoItems: []
};

TodoList.propTypes = {
  todoItems: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      completed: PropTypes.bool,
      index: PropTypes.number
    })
  ),
  onTodoItemClick: PropTypes.func
};

export default TodoList;
