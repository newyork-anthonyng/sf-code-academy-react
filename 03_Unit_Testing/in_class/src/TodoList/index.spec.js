import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import TodoList from "./";

Enzyme.configure({ adapter: new Adapter() });

// Add new tests here
it("should return true", function() {
  expect(true).toEqual(true);
});

const defaultProps = {
  todoItems: [
    { text: "One", completed: true },
    { text: "Two", completed: true },
  ],
  onTodoItemClick: function() {}
};

it("should render correctly with todoItems", function() {
  const wrapper = shallow(
    <TodoList {...defaultProps} />
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render correctly when todoItems is undefined", function() {
  const wrapper = shallow(
    <TodoList {...defaultProps} todoItems={undefined} />
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
