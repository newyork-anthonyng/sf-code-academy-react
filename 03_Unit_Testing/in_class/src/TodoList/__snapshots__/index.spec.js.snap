// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when todoItems is undefined 1`] = `<ul />`;

exports[`should render correctly with todoItems 1`] = `
<ul>
  <TodoItem
    completed={true}
    index={0}
    key="0"
    onClick={[Function]}
    text="One"
  />
  <TodoItem
    completed={true}
    index={1}
    key="1"
    onClick={[Function]}
    text="Two"
  />
</ul>
`;
